nixos/modules/image/repart-verity-store.nix

```diff
@@ -0,0 +1,198 @@
+# opinionated module that can be used to build nixos images with
+# a dm-verity protected nix store and a volatile rootfs
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.image.repart.verityStore;
+
+  verityMatchKey = "store";
+
+  # TODO: make these and other arch mappings available from systemd-lib for example
+  partitionTypes = {
+    usr =
+      {
+        "x86_64" = "usr-x86-64";
+        "arm64" = "usr-arm64";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+
+    usr-verity =
+      {
+        "x86_64" = "usr-x86-64-verity";
+        "arm64" = "usr-arm64-verity";
+      }
+      ."${pkgs.stdenv.hostPlatform.linuxArch}";
+  };
+
+  ukiPath = "/EFI/Linux/${config.system.boot.loader.ukiFile}";
```

Comment:
This should maybe be a config option defaulting to this value, e.g. for direct UKI boot